firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -135,12 +211,71 @@ public UpdateTask updateApp() {
     return (UpdateTask) Tasks.forResult(new UpdateState(0, 0, UpdateStatus.PENDING));
   }
 
-  private boolean supportsCustomTabs(Context context) {
-    Intent customTabIntent = new Intent("android.support.customtabs.action.CustomTabsService");
-    customTabIntent.setPackage("com.android.chrome");
-    List<ResolveInfo> resolveInfos =
-        context.getPackageManager().queryIntentServices(customTabIntent, 0);
-    return resolveInfos != null && !resolveInfos.isEmpty();
+  /** Returns true if the App Distribution tester is signed in */
+  @NonNull
+  public boolean isTesterSignedIn() {
+    return false;
+  }
+
+  /** Signs out the App Distribution tester */
+  public void signOutTester() {}
+
+  @Override
+  public void onActivityCreated(
+      @NonNull @NotNull Activity activity, @androidx.annotation.Nullable @Nullable Bundle bundle) {
+    Log.d(TAG, "Created activity: " + activity.getClass().getName());
+    // if signinactivity is created, sign-in was succesful
+    if (currentlySigningIn && activity instanceof SignInResultActivity) {
+      currentlySigningIn = false;
+      signInTaskCompletionSource.setResult(null);
+    }
+  }
+
+  @Override
+  public void onActivityStarted(@NonNull @NotNull Activity activity) {
+    Log.d(TAG, "Started activity: " + activity.getClass().getName());
+  }
+
+  @Override
+  public void onActivityResumed(@NonNull @NotNull Activity activity) {
+    Log.d(TAG, "Resumed activity: " + activity.getClass().getName());
+
+    // signInActivity is only opened after successful redirection from signIn flow,
+    // should not be treated as reentering the app
+    if (activity instanceof SignInResultActivity) {
+      return;
+    }
+
+    // throw error if app reentered during signin
+    if (currentlySigningIn) {
+      currentlySigningIn = false;
```

Comment:
nit: this should be an AUTHENTICATION_CANCELLED error instead of AUTHENTICATION_FAILURE